src/dawn/Optimizer/PassDataLocalityMetric.cpp

```diff
@@ -125,13 +128,23 @@ class ReadWriteCounter : public ASTVisitorForwarding {
     if(cache.getCacheType() == Cache::K) {
       if(cache.getCacheIOPolicy() == Cache::fill ||
          cache.getCacheIOPolicy() == Cache::fill_and_flush) {
-        std::cout << "   Fill: " << getNameFromAccessID(AccessID) << std::endl;
+//        std::cout << "   Fill: " << getNameFromAccessID(AccessID) << std::endl;
         numReads_++;
+        auto iter = individualReadWrites_.find(AccessID);
```

Comment:
just wondering, would the one line
```
individualReadWrites_[AccessID].numReads++;
```
do the whole magic? individualReadWrites_[AccessID] would default construct an element if key is not found...

Reply:
This is actually really smart :+1: